src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -65,6 +65,16 @@ internal static class ProtocolConversions
             { WellKnownTags.NuGet, LSP.CompletionItemKind.Text }
         };
 
+        // TO-DO: More LSP.CompletionTriggerKind mappings are required to properly map to Roslyn CompletionTriggerKinds.
+        // https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1178726
+        public static Completion.CompletionTriggerKind LSPToRoslynCompletionTriggerKind(LSP.CompletionTriggerKind triggerKind)
+            => triggerKind switch
+            {
+                LSP.CompletionTriggerKind.Invoked => Completion.CompletionTriggerKind.Invoke,
+                LSP.CompletionTriggerKind.TriggerCharacter => Completion.CompletionTriggerKind.Insertion,
+                _ => throw new ArgumentException($"LSP.CompletionTriggerKind {triggerKind} unexpected.")
```

Comment:
I don't know if we should necessarily throw but instead log something and continue.  It's possible that new trigger kinds get added underneath us (for example, the spec defines a kind not listed here, TriggerForIncompleteCompletions)